src/mame/video/powervr2.cpp

```diff
@@ -2861,7 +2861,10 @@ void powervr2_device::render_to_accumulation_buffer(bitmap_rgb32 &bitmap, const
 
 	dc_state *state = machine().driver_data<dc_state>();
 	address_space &space = state->m_maincpu->space(AS_PROGRAM);
-	uint32_t c=space.read_dword(0x05000000+((isp_backgnd_t & 0xfffff8)>>1)+(3+3)*4);
+
+	// TODO: read ISP/TSP command from isp_background_t instead of assuming gourad-shaded
+	// full-screen polygon.
+	uint32_t c=space.read_dword(0x05000000+(param_base & 0xffffff)+((isp_backgnd_t & 0xfffff8)>>1)+(3+3)*4);
```

Comment:
1. `Gouraud-shaded`, as the [inventor surname](https://en.wikipedia.org/wiki/Henri_Gouraud_(computer_scientist))
2. Is `param_base` supposed to be masked with ~7 too? That and `isp_backgnd_t` should be done in the `_w `write accessors instead of here for optimization.

Reply:
1. Oops lol.
2. Huh? param_base and isp_background_t are both 32-bit registers inside variables. Masking in the _w write call doesn't make sense. Also this is read once per frame to do an incorrect sofware rectangle fill on a video core that struggles to hit 15fps on modern computers. It doesn't matter if its done once in the write call or once in the read call. I guess it doesn't matter if param_base is masked here as IIRC nothing but an address is written to it.

Reply:
It matters for readability, accuracy and for a very minor optimization. They are set only by the _w fns, so making to mask once per frame HERE and not in other places conceals a possible bug if the SW tries to mask with undefined bits.